calculate_number_of_profiled_coding_base_pairs/add_number_of_profiled_coding_base_pairs.py

```diff
@@ -0,0 +1,94 @@
+import csv
+import pandas as pd
+import os
+import sys
+import argparse
+import ntpath
+
+def generate_dict(key, value, dictionary):
+    if not pd.isnull(key):
+        if ';' not in key:
+            dictionary[key] = value
+        else:
+            genes = key.split('; ')
+            for gene in genes:
+                dictionary[gene] = value
+
+def update_file(args):
+    inputFile = args.inputFile
+    uniprotFile = args.uniprotFile or os.getcwd() + "/uniprot_gene_name_with_protein_length.tab"
+    outputFile = args.outputFile or os.path.dirname(inputFile) + "/output_" + ntpath.basename(inputFile)
```

Comment:
same with `os.path.dirname(inputFile)`. If an absolute path and name for the output file was not defined, the output file tend to get generated under the `\` root folder becuase `os.path.dirname(inputFile)` was get back blank when I run the script.

Reply:
@yichaoS It works fine on my side, `os` seems to be a very basic package, I have no clue why it doesn't work for you... So how do you usually read path and direction name? Could you give me some suggestions?

Reply:
@leexgh the package itself is working. I think it's because I'm running the script under a different path, which is not where the scripts are, so there's no complete/absolute path for the input file, therefore `os.path.dirname()` is outputting blank. For example, my cmd was `python ~/data/datahub-study-curation-tools/calculate_number_of_profiled_coding_base_pairs/add_number_of_profiled_coding_base_pairs.py -u ~/data/datahub-study-curation-tools/calculate_number_of_profiled_coding_base_pairs//uniprot_gene_name_with_protein_length.tab -i data_gene_panel_AmpliSeq.txt` and the error is` IOError: [Errno 13] Permission denied: '/output_data_gene_panel_AmpliSeq.txt'`